clients/src/main/java/org/apache/kafka/clients/consumer/internals/events/ApplicationEventProcessor.java

```
@@ -198,23 +198,47 @@ private void process(final CreateFetchRequestsEvent event) {
     }
 
     private void process(final AsyncCommitEvent event) {
-        if (!requestManagers.commitRequestManager.isPresent()) {
+        if (requestManagers.commitRequestManager.isEmpty()) {
+            event.future().complete(Map.of());
```

Should we better complete exceptionally? I expect this would only happen during development phase if a bug is introduced, but still, it would be easier to catch if we make the app event fail because the expected manager is not present. I notice there's no consistency in how we handle these (some process calls complete exceptionally, others just return/complete, but I would lean towards not swallowing the unlikely/unexpected error here)

---

Thanks for the suggestion. There are some `process` don't align with the rule. Do we want to create a Jira to handle it? Or I can update them in this PR?

---

uhm I would leave it out of this PR. Even though there are some other process calls that should probably fail if the expected manager is not present, in some we shouldn't fail and just return/no-op as they do now (ie. commitOnClose no-op if no commit mgr, triggered from close regardless of groupId present or not).